# VISUALIZATION OF CONDENSED MULTIPLE SEQUENCE ALIGNEMENTS AND PHYLOGENETIC TREES

### List of participants and affiliations:
- Christian Zmasek (czmasek@jcvi.org) - Team leader
- Chrissy Aceves (caceves@scripps.edu) 
- Nate Matteson (natem@scripps.edu)
- Grace Nabakooza (sxv8@cdc.gov)
- Daniel Chen (dchen32@uw.edu)
- Diana Ir (diana.ir@state.co.us) 


## Background 
Phylogenetic visualization of large multiple sequence alignments is difficult. Since phylogenies are reconstructed based on differences in a multiple sequence alignment (MSA), tools that utilize only the positions with variation in the MSA could improve phylogenetic reconstruction and visualization.

## Project Goals
 We developed a drag and drop interface where the user provides (1) a vcf file and (2) a newick tree file and render a visualization focusing on single nucleotide variants (SNVs). 
 

## Prerequisites
Python 3

## Installation

## Input

## Approach
- Possibly write the backend in Python, the frontend in JS, and tie it together in a flask app?


## Results

## Future Work
test
